dev/PackageManager/API/PackageManager.idl

```diff
@@ -470,8 +471,10 @@ namespace Microsoft.Windows.Management.Deployment
         //-------------------------------------------------------------
         // IsRegistrationPending
 
+        [contract(PackageDeploymentContract, 2)]
```

Comment:
IsPackageRegistrationPending and IsPackageRegistrationPendingForUser are present in 1.5-stable. The only difference I see is in their parameter names. IIRC parameter names can be significant, though I think that was mostly for out-param names (affects how they project for languages like Javascript which can have multiple return values).
In particular, I don't know offhand whether param names are considered in the algorithm that hashes together signature elements of the interface to generate an IID.

My recommendation would be to remove the contract attribute, and revert the param name to packageFamilyName just in case. If you want to validate that it has no impact on IID, it would probably be safe to change.

Reply:
Related: http://task.ms/53280356

Reply:
And to be clear: adding this contract attribute will _remove_ these methods from the existing IPackageDeploymentManager interface, and change its IID. So this would be a breaking change for 1.5 -> 1.6

Reply:
You're right. I misread the diff

Param name doesn't affect IID generation, right? Just method name, return values and parameter types?

Reply:
I _think_ you're right. It's a MidlRT implementation detail, but I would expect node_param::GetInterfaceString to be the thing that contributes to the IID generation, and it doesn't factor in its own name, it just defers to the child node (which should represent the param type). Disclaimer: I'm a bit rusty in MidlRT code, and would still recommend building this and comparing the IID in the generated WinMD against an existing 1.5 WinMD.

Reply:
Though per the bug that @codendone linked, parameter name changes are breaking for named parameters in C#. Obviously breaking the IID is a bigger deal, but now's the right time to decide whether correcting inaccurate param names is worth the minor break.

Reply:
#Sad now. OK, param names going back until the future mythical 2.0 :P